### ISSUE TITLE
modprobe vboxsf fails, virtualbox-guest-additions

### ISSUE THREAD

**Issue by eadwu** [NixOS/nixpkgs#47061]
## Issue description
Attempt to install virtualbox-guest-additions. Tried `linuxPackages_latest` and `linuxPackages_testing` and also virtualbox v5.2.18.
`dmesg | grep vbox`
```
[    0.885435] stage-1-init: loading module vboxsf...
[    0.887157] vboxsf: loading out-of-tree module taints kernel.
[    0.887182] vboxsf: Unknown symbol VBoxGuest_RTMemTmpFree (err 0)
[    0.887183] vboxsf: Unknown symbol VBoxGuestIDC (err 0)
[    0.887185] vboxsf: Unknown symbol VBoxGuest_RTSemFastMutexRequest (err 0)
[    0.887189] vboxsf: Unknown symbol VBoxGuest_RTSemFastMutexRelease (err 0)
[    0.887190] vboxsf: Unknown symbol VBoxGuest_RTLogRelGetDefaultInstanceEx (err 0)
[    0.887192] vboxsf: Unknown symbol VBoxGuest_RTStrCopy (err 0)
[    0.887193] vboxsf: Unknown symbol VBoxGuest_RTErrConvertToErrno (err 0)
[    0.887197] vboxsf: Unknown symbol VBoxGuest_RTSemFastMutexCreate (err 0)
[    0.887198] vboxsf: Unknown symbol VBoxGuest_RTSemFastMutexDestroy (err 0)
[    0.887204] vboxsf: Unknown symbol VBoxGuest_RTAssertShouldPanic (err 0)
[    0.887206] vboxsf: Unknown symbol VBoxGuest_RTLogLoggerEx (err 0)
[    0.887208] vboxsf: Unknown symbol VBoxGuest_RTMemTmpAllocTag (err 0)
[    0.887211] vboxsf: Unknown symbol VBoxGuest_RTAssertMsg1Weak (err 0)
[    0.887214] vboxsf: Unknown symbol VBoxGuest_RTAssertMsg2Weak (err 0)
[    0.900375] stage-1-init: modprobe: ERROR: could not insert 'vboxsf': Unknown symbol in module, or unknown parameter (see dmesg)
[   13.594264] vboxdrv: Found 12 processor cores
[   13.613084] vboxdrv: TSC mode is Invariant, tentative frequency 2207999007 Hz
[   13.613085] vboxdrv: Successfully loaded version 5.2.18 (interface 0x00290001)
[   17.545456] IPv6: ADDRCONF(NETDEV_UP): vboxnet0: link is not ready
```

### Steps to reproduce
```nix
{
  virtualisation = {
    virtualbox = {
      guest = {
        enable = true;
      };
    };
  };
}
```

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.18.9, NixOS, 19.03.git.d5e99fa (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1pre6377_954d1f4d`
 - channels(root): `"nixos-19.03pre152634.218ce4de508"`
 - nixpkgs: `/home/xps/Downloads/nixpkgs`
```

**Comment by eadwu** [comment]
Actually seems to be caused by a problem in my configuration, I'll close for now and post the problem if I ever get around to looking for it.

**Comment by CMCDragonkai** [comment]
I met this error now. @eadwu did you find out what caused this?

**Comment by eadwu** [comment]
For me, I was running `virtualisation.virtualbox.guest.enable = true` on my host machine which doesn't really make sense so I changed it to `virtualisation.virtualbox.host.enable = true` while `virtualisation.virtualbox.guest.enable = true` was set in my VM, like in here, [in my gitlab](https://gitlab.com/eadwu/nixos-configuration/blob/master/profiles/vm.nix), where it is enabled in [nixos/modules/virtualisation/virtualbox](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/virtualisation/virtualbox-image.nix#L102).

**Comment by CMCDragonkai** [comment]
I'm definitely using a Virtualbox guest and set the guest enable and still got this error.

**Comment by eadwu** [comment]
Are you able to use the functionality given by the guest additions regardless of the error?

**Comment by RogerWeihrauch** [comment]
howdy
I just ran "/mnt/VBoxLinuxAdditions.run" after mounting as root and got the same errors shown with dmesg. Seems modprobe does also not work: --- snip ---
VirtualBox Guest Additions: Building the VirtualBox Guest Additions kernel modules.  This may take a while.
update-initramfs: Generating /boot/initrd.img-4.18.0-0.bpo.1-amd64
VirtualBox Guest Additions: Running kernel modules will not be replaced until the system is restarted
VirtualBox Guest Additions: Starting.
VirtualBox Guest Additions: modprobe vboxsf failed --- snap ---
I noticed this swapping phenomenon on an earlier VBox releases already: that the vboxsf module was (sometimes) not (always) included/compiled/running after VBoxLinuxAdditions.run. Someone some clue on this?
Your help is very welcome.
Kind regards,
Roger
